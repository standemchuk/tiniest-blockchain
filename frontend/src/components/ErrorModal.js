import React from 'react'
import Button from '@material-ui/core/Button'
import Modal from '@material-ui/core/Modal'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

const top = 50 + Math.round(Math.random() * 20) - 10
const left = 50 + Math.round(Math.random() * 20) - 10

const styles = theme => ({
  modal: {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
    position: 'absolute',
    width: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing.unit * 4
  },
  buttonWrapper: {
    marginTop: 10,
    display: 'flex',
    justifyContent: 'space-between'
  }
})

const ErrorModal = (props) => (
  <Modal open={props.open} onClose={props.handleModalClose} >
    <div className={props.classes.modal}>
      <Typography
        variant='title'
        id='modal-title'
        aria-labelledby='simple-modal-title'
        aria-describedby='simple-modal-description'
      >
        Something went wrong
      </Typography>
      <Typography variant='subheading' id='simple-modal-description'>
        There was a problem with retrieving the data you've requested
      </Typography>
      <div className={props.classes.buttonWrapper}>
        <Button variant={'outlined'} color={'secondary'} onClick={props.handleRetry}>Retry</Button>
        <Button variant={'outlined'} color={'primary'} onClick={props.handleModalClose}>Ignore</Button>
      </div>
    </div>
  </Modal>
)

export default withStyles(styles)(ErrorModal)
