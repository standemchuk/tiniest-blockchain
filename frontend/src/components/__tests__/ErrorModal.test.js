import React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { ErrorModal } from '../ErrorModal'

it('renders correctly', () => {
  const tree = createMount()(
    <ErrorModal
      open
      handleModalClose={jest.fn()}
      handleRetry={jest.fn()}
      classes={{ modal: 'modal', buttonWrapper: 'buttonWrapper' }}
    />)
  expect(tree.html()).toMatchSnapshot()
})
