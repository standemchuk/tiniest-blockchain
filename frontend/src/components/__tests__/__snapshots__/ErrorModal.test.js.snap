// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div class=\\"MuiModal-root-1\\"><div class=\\"MuiBackdrop-root-3\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"modal\\" role=\\"document\\" tabindex=\\"-1\\"><h2 class=\\"MuiTypography-root-5 MuiTypography-title-11\\" id=\\"modal-title\\" aria-labelledby=\\"simple-modal-title\\" aria-describedby=\\"simple-modal-description\\">Something went wrong</h2><h3 class=\\"MuiTypography-root-5 MuiTypography-subheading-12\\" id=\\"simple-modal-description\\">There was a problem with retrieving the data you've requested</h3><div class=\\"buttonWrapper\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-53 MuiButton-root-29 MuiButton-text-31 MuiButton-textSecondary-33 MuiButton-outlined-37\\" type=\\"button\\"><span class=\\"MuiButton-label-30\\">Retry</span><span class=\\"MuiTouchRipple-root-56\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-53 MuiButton-root-29 MuiButton-text-31 MuiButton-textPrimary-32 MuiButton-outlined-37\\" type=\\"button\\"><span class=\\"MuiButton-label-30\\">Ignore</span><span class=\\"MuiTouchRipple-root-56\\"></span></button></div></div></div>"`;
